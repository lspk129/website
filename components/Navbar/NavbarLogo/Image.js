import styled from 'styled-components';

const Image = styled.img`
  width: 100px;
  padding-top: 10px;
`;

export default Image;
