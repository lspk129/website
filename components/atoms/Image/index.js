import styled from 'styled-components';

const Image = styled.img`
  ${({ width }) => width && `width: ${width}px`}
`;

export default Image;
