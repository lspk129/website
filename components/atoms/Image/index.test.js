import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Image from './';

describe('Image', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Image />);
    expect(shallow(<Image />)).toBePresent();
    expect(shallow(<Image />)).toHaveTagName('img');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with width props', () => {
    const wrapper = shallow(<Image width={300} />);
    expect(wrapper).toHaveProp('width', 300);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
