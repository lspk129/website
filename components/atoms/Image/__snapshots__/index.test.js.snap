// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders correctly 1`] = `
<img
  className="sc-bdVaJa ibTDPh"
/>
`;

exports[`Image renders with width props 1`] = `
<img
  className="sc-bdVaJa fbrXsZ"
  width={300}
/>
`;
